engine-nextgen/src/main/kotlin/graphql/nadel/validation/NadelHydrationValidation.kt

```diff
@@ -65,6 +66,23 @@ internal class NadelHydrationValidation(
                 continue
             }
 
+            val serviceRequiresActorFieldsToBeDeclaredInOverallSchema =
+                nadelValidationHints?.requireHydrationActorFieldInOverallSchemaHint?.getHintValue(actorService) ?: false
+            if (serviceRequiresActorFieldsToBeDeclaredInOverallSchema) {
+                val actorFieldFromOverall = overallSchema.queryType.getFieldAt(hydration.pathToActorField)
```

Comment:
Just double checking: 
- I thought hydration instructions used the underlying schema as a reference (as in, the hydration query is sent directly to the underlying service, so it should be valid against that schema), so it didn't really matter if the actor field exists or not in the overall, is that still the case (or maybe it never was the cause and I'm dreaming 😬 ).

Reply:
definitely has been the case so far but we are changing it now
we want actor fields in the overall schema so we can get `@CloudId` and `@ARI` from arguments for hydration queries. also we believe that after some additional refactoring we will be able to simplify the code when we don't need to be constantly switch between overall and underlying schemas

Reply:
Awesome! Thanks for confirming ;-)